src/pages/Reset.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from "react";
+import axios from "axios";
+import { Formik, ErrorMessage } from 'formik';
+import * as Yup from 'yup';
+import { Navigation } from "../components/Navigation";
+import { useNavigate } from "react-router-dom";
+import { LoadingSpinner } from "../components/LoadingSpinner";
+
+const validationSchema = Yup.object().shape({
+  email: Yup.string().required('Email jest wymagany')
+});
+
+export const ResetPassword = () => {
+  const navigate = useNavigate();
+  const [isLoading, setLoading] = useState(false);
+
+  const submit = async (values, { setStatus }) => {
+    const emailData = {
+      email: values.email
+    };
+    try {
+      setLoading(true);
+      // Wysyłanie adresu email na serwer
+      const { data } = await axios.post('/reset-password', emailData);
+
+      if (!data) {
+        console.log('Nie udało się wysłać emaila');
+        setStatus('Nie udało się wysłać emaila');
+        return;
+      }
+
+      // Przekierowanie do strony głównej
+      navigate('/');
+    } catch (error) {
+      if (error?.response?.status === 400) {
+        setStatus('Nie udało się wysłać emaila.');
+        return;
+      }
+      setStatus('Wystąpił błąd. Spróbuj ponownie.');
+    } finally {
```

Comment:
Nitpicking: w tym przypadku 400 to nic innego jak podano niepoprawny mail, ale zazwyczaj nie informuje się użytkownika, czy podał poprawny mail, tylko jest komunikat w stylu "Jeżeli podano poprawny mail, to została wysłana wiadomość mail z linkem do zmiany hasła".